src/rust/engine/fs/src/store.rs

```diff
@@ -1019,6 +1052,48 @@ mod local {
     pub fn all_lmdbs(&self) -> Vec<(Arc<Environment>, Database, Database)> {
       self.lmdbs.values().cloned().collect()
     }
+
+    pub fn compact(&self) -> Result<(), String> {
+      for b in 0x00..0x10 {
+        let key = b << 4;
+
+        let mut dirname = {
```

Comment:
Worth adding a tempdir dep? Alternatively, should this fail cleanly if the `.tmp` directory already exists with something like "compaction already in progress"?

Reply:
Done